src/io/csv/csv-reader.cu

```diff
@@ -0,0 +1,1232 @@
+				int bitIdx		= whichBit(rec_id + col);		// which bit - over an 8-bit index
+				setBit(valid[col]+bitmapIdx, bitIdx);		// This is done with atomics
+
+			// }
+		}
+		else if(dtype[col]==gdf_dtype::GDF_STRING){
+			str_cols[stringCol][rec_id].first 	= NULL;
+			str_cols[stringCol][rec_id].second 	= 0;
+			stringCol++;
+			if(threadIdx.x==0&&blockIdx.x==0){
+				printf("^^^^ %d %ld 0 \n", col, start);	
+			}
+
+		}
+
+		pos++;
+		start=pos;
+		col++;	
+	}
+}
+
+__device__
+void removePrePostWhiteSpaces2(char *data, long* start_idx, long* end_idx) {
+	while(*start_idx < *end_idx && data[*start_idx] == ' ')
+		*start_idx=*start_idx+1;
+	while(*start_idx < *end_idx && data[*end_idx] == ' ')
+		*end_idx=*end_idx-1;
+}
+
+
+/*
+ * Return which bit is set
+ * x is the occurrence: 1 = first, 2 = seconds, ...
+ */
+__device__ int findSetBit(int tid, long num_bits, uint64_t *r_bits, int x) {
+
+	int idx = tid;
+
+	if ( x == 0 )
+		return -1;
+
+	int withinBitCount = 0;
+	int offset = 0;
+	int found  = 0;
+
+	uint64_t bitmap = r_bits[idx];
+
+	while (found != x)
+	{
+		if(bitmap == 0)
+		{
+			idx++;
+			if (idx >= num_bits)
+				return -1;
+			bitmap = r_bits[idx];
+			offset += 64;
+			withinBitCount = 0;
+		}
+
+		if ( bitmap & 1 ) {
+			found++;			//found a set bit
+		}
+
+		bitmap >>= 1;
+		++withinBitCount;
+	 }
+
+	offset += withinBitCount -1;
+
+
+	return offset;
+}
+
+
+
+
+
+//---------------------------------------------------------------------------------------------------------------
+//
+//			Debug functions below this point
+//
+//---------------------------------------------------------------------------------------------------------------
+
+void printCheck(raw_csv_t * csvData, int start_data_idx, int num_blocks, const char * text) {
+
+	cudaDeviceSynchronize();
+
+	std::cout << "\n--------------------------------\n";
+	std::cout << "Checking (dependent on Unified Memory) - " 		<< text << std::endl;
+	std::cout << "\tNumber of Bytes   - " << csvData->num_bytes 	<< std::endl;
+	std::cout << "\tNumber of Bitmaps - " << csvData->num_bits  	<< std::endl;
+	std::cout << "\tNumber of Records - " << csvData->num_records  	<< std::endl;
+
+
+	char * data = csvData->data;
+
+	int data_idx 	= start_data_idx;
+
+	if ( data_idx != 0)
+		while ( data_idx % 64 != 0 )
+			data_idx++;
+
+	std::cout << "\tStarting Index specified - " << start_data_idx << "  adjusted to " << data_idx << std::endl;
+
+	int bitmap_idx  = data_idx / 64;
+
+	std::cout << "\tStarting Bitmap Index - " << bitmap_idx  << std::endl;
+	std::cout << "[data, bit] =>                 64 bytes of data                 \t rec_bit   field_bits    Record counts"  << std::endl;
+
+
+	for ( int loop = 0; loop < num_blocks; loop++) {
+		std::cout << "[" << std::setw(6) << data_idx << " ,  " << std::setw(6) << bitmap_idx << "] =>  ";
+
+		 for ( int x = 0; x < 64; x++) {
+
+			if ( data_idx < csvData->num_bytes) {
+				if (data[data_idx] == '\n')
+					std::cout << "\033[1m\033[31mNL\033[0m ";
```

Comment:
This may be a question for @kkraus14 et al., but is it OK to use ANSI escape sequences here? That doesn't seem like it would be supported on all platforms.

Reply:
@BradReesWork Does the CSV reader support UTF-8 or is it ANSI only right now?

Reply:
UTF-8.  That line is in a debug block that is being removed

Reply:
that line was used to output the input data and highlight in a different color the breaks